### PR TITLE
Large reorg and cleanup

### PR DESCRIPTION
**Pull request by TimOrme**
Fixes some issues with interpolated values in regex as well.
This will set up for some follow up PR's to add tests and corrections
for date parsing issues, as well as mypy integration.

Removed lots of code that we don't use a production env, mostly
around evaluation of performance and some more proprietary features
like the Standford NER.

Apologies for the size of the PR.  Most of the changes here are purely organizational, and this wont be released until we have a more robust test suite and type checking.

### CODE REVIEW

File: philter_lite/asterisk.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+from .coordinate_map import CoordinateMap
+
+
+def save_to_asterisk(contents, output_file):
+    with open(output_file, "w", encoding="utf-8", errors="surrogateescape") as f:
+        f.write(contents)
+
+
+def transform_text_asterisk(txt, include_map: CoordinateMap):
```
Comment (skucherlapati) [inline philter_lite/asterisk.py:11]: is this to make sure we are preserving punctuations within the phi text?

Comment (TimOrme) [reply philter_lite/asterisk.py:11]: No, `asterisk` is just one of the output formats the `philter` supports, replacing PHI with asterisks.  Basically, given some text like:

```
My birthdate is 01/01/1990
```

The asterisk format would be:

```
My birthdate is **********
```

File: philter_lite/filters/__init__.py [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+        "regex_context",
+        "stanford_ner",
+        "pos_matcher",
+        "match_all",
+    }
+
+    filter_type = filter_dict["type"]
+
+    if filter_type not in known_pattern_types:
+        raise Exception("Pattern type is unknown", filter_type)
```
Comment (skucherlapati) [inline philter_lite/filters/__init__.py:75]: what is the difference between raising a value error vs a base exception?

**Review by skucherlapati — APPROVED** [review]

File: philter_lite/filters/__init__.py (hunk 2 above)
Comment (TimOrme) [reply philter_lite/filters/__init__.py:75]: ValueError is probably better here, though not much practical difference.  I think its unlikely that someone would want to handle these exceptions either way, but I can change it.

**Pull request merged**